public/js/controllers/charts_controller.js

```diff
@@ -16,25 +16,26 @@ let Dygraph // lazy loaded on connect
 const aDay = 86400 * 1000 // in milliseconds
 const aMonth = 30 // in days
 const atomsToDCR = 1e-8
-const windowScales = ['ticket-price', 'pow-difficulty', 'missed-votes']
-const lineScales = ['ticket-price']
+const chartsToHideBin = ['ticket-price', 'pow-difficulty', 'missed-votes']
+const chartsToHideScale = ['ticket-price']
+const chartsToHideAll = ['duration-btw-blocks']
 // index 0 represents y1 and 1 represents y2 axes.
 const yValueRanges = { 'ticket-price': [1] }
-var ticketPoolSizeTarget, premine, stakeValHeight, stakeShare
+const altXLabel = { 'duration-btw-blocks': 'Duration Between Blocks (Seconds)' }
+const chartsToShowVisibility = {
+  'ticket-price': { 'option1': 'Price', 'option2': 'Tickets Bought' },
+  'duration-btw-blocks': { 'option1': 'Actual Count', 'option2': 'Expected Count' }
+}
+var ticketPoolSizeTarget, premine, stakeValHeight, stakeShare, sum
 var baseSubsidy, subsidyInterval, subsidyExponent, windowSize, avgBlockTime
 var rawCoinSupply, rawPoolValue
 
-function usesWindowUnits (chart) {
-  return windowScales.indexOf(chart) > -1
-}
-
-function isScaleDisabled (chart) {
-  return lineScales.indexOf(chart) > -1
-}
-
-function intComma (amount) {
-  return amount.toLocaleString(undefined, { maximumFractionDigits: 0 })
-}
+var isBinDisabled = (chart) => chartsToHideBin.indexOf(chart) > -1
+var isScaleDisabled = (chart) => chartsToHideScale.indexOf(chart) > -1
+var isAllControlsDisabled = (chart) => chartsToHideAll.indexOf(chart) > -1
+var customXLabel = (chart) => altXLabel[chart] || ''
```

Comment:
This comment applies to the general approach you've taken to handling the new axis here and below. By my count, you've added 10s of new functions to handle 1 new button/axis unit. I see that there is some special handling to be done for the new chart because of the new type of axis, but this approach is a bit tedious. Can't you just set the button visibility and null settings in the switch/case in `plotGraph`?

I'm happy to work with you towards a better solution here. Let me know if I can help.